tree/30_generic_methods/_classes_aggregate_ifthenelse.cf

```diff
@@ -0,0 +1,72 @@
+#####################################################################################
+# Copyright 2013 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Class aggregate ifthenelse
+# @description Aggregate class outcomes within an if then else structure
+#
+# @parameter test_prefix        The prefix of classes comming from the test method
+# @parameter then_prefix        The prefix of classes comming from the method called in the 'then' case
+# @parameter else_prefix        The prefix of classes comming from the method called in the 'else' case
+# @parameter destination_prefix The prefix of the destination class
+#
+# @class_prefix 
+# @class_parameter 
+# This bundle will define a class ${destination_prefix}_{kept,repaired,not_ok,ok,reached}
+# based on  the test result and the method called afterwards.
+#
+# If the if the 'if' call has a '_true' outcome then the 'then' result is copied to destination
+# If the if the 'if' call has a '_false' outcome then the 'else' result is copied to destination
+# 
+# 'then_prefix' and 'else_prefix' can be replaced by the litteral string "success" or "failure"
+# In such case, the outcome is respectively replaced by a call to _classes_success or _classes_failure
+
+bundle agent _classes_aggregate_two(test_prefix, then_prefix, else_prefix, destination_prefix)
+{
+  classes:
+      "test_true"   expression => "${test_prefix}_true";
+      "test_false"  expression => "${test_prefix}_false";
+
+      "then_success"  expression => strcmp("success", "${then_prefix}");
+      "then_failure"  expression => strcmp("failure", "${then_prefix}");
```

Comment:
this doesn't look correct: you are looking for success and failure in then_prefix and else_prefix
so they can't be anything else than these values, killing the concept of putting the class prefix we want